lisa/src/main/java/it/unive/lisa/analysis/nonrelational/value/BaseNonRelationalValueDomain.java

```diff
@@ -143,6 +143,12 @@ public final T eval(ValueExpression expression, ValueEnvironment<T> environment,
 			if (right.isBottom())
 				return right;
 
+			if (binary.getOperator() == BinaryOperator.TYPE_CAST)
```

Comment:
All good, but the same reasoning should be applied to `BaseInferredValue` and `InferredTypes` should be changed accordingly.